extensions/logging/README.md

```diff
@@ -0,0 +1,158 @@
+
+## Stability: :warning:Experimental:warning:
+
+> Experimental packages provide early access to advanced or experimental
+> functionality to get community feedback. Such modules are published to npm
+> using `0.x.y` versions. Their APIs and functionality may be subject to
+> breaking changes in future releases.
+
+## Architecture overview
+
+![logging-component](logging-component.png)
+
+## Installation
+
+```sh
+npm install --save @loopback/extension-logging
+```
+
+## Basic use
+
+The component should be loaded in the constructor of your custom Application
+class.
+
+Start by importing the component class:
+
+```ts
+import {LoggingComponent} from '@loopback/extension-logging';
+```
+
+In the constructor, add the component to your application:
+
+```ts
+this.component(LoggingComponent);
+```
+
+Now your application can add a controller as follows to leverage the logging
+facilities:
+
+```ts
+import {inject} from '@loopback/context';
+import {Logger} from '@loopback/extension-logging';
+
+class MyController {
+  // Inject a winston logger
+  @inject(LoggingBindings.WINSTON_LOGGER)
+  private logger: Logger;
+
+  // http access is logged by a global interceptor
+  @get('/greet/{name}')
+  // log the `greet` method invocations
+  @log()
+  greet(@param.path.string('name') name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @get('/hello/{name}')
+  hello(@param.path.string('name') name: string) {
+    // Use the winston logger explicitly
+    this.logger.log('info', `greeting ${name}`);
+    return `Hello, ${name}`;
+  }
+}
+```
+
+## Configure the logging component
+
+The component contributes bindings with keys declared in `LoggingBindings`
+namespace below:
+
+- FLUENT_SENDER - A fluent sender
+- WINSTON_LOGGER - A winston logger
+- WINSTON_TRANSPORT_FLUENT - A fluent transport for winston
+- WINSTON_INTERCEPTOR - A local interceptor set by `@log` to log method
+  invocations
+- WINSTON_ACCESS_LOGGER - A global interceptor that logs http access with
+  [Morgan](https://github.com/expressjs/morgan) format
+
+The fluent sender and transport for winston can be configured against
+`FLUENT_SENDER`:
+
+```ts
+import {LoggingBindings} from '@loopback/extension-logging';
+
+app.configure(LoggingBindings.FLUENT_SENDER).to({
+  host: process.env.FLUENTD_SERVICE_HOST || 'localhost',
+  port: +(process.env.FLUENTD_SERVICE_PORT_TCP || 0) || 24224,
+  timeout: 3.0,
+  reconnectInterval: 600000, // 10 minutes
+});
+```
+
+The winston logger can be configured against `LoggingBindings.WINSTON_LOGGER`:
+
+```ts
+import {LoggingBindings} from '@loopback/extension-logging';
+
+ctx.configure<LoggerOptions>(LoggingBindings.WINSTON_LOGGER).to({
+  level: 'info',
+  format: format.json(),
+  defaultMeta: {framework: 'LoopBack'},
+});
+```
+
+The winston logger accepts two types of extensions to the following extension
+points:
+
+- WINSTON_TRANSPORT = 'logging.winston.transport'
+- WINSTON_FORMAT = 'logging.winston.format'
+
```

Comment:
I am missing a high-level summary of what is extensible and what is hard-coded, what is provided out-of-the-box and what requires explicit opt-in/setup.

IIUC:

- Winston logger API is hard-coded 🤷‍♂
- It's possible to provide a custom Winston logger instance (🤔 why would anybody want to do that?)
- It's possible to register multiple transports 👍 
- It's possible to register multiple formatters 👍 

We should also answer the following questions
- What formatters are registered by default? How can I configure them? How to disable them?
- What transports are registered by default? How can I configure them? How to disable them?
- What logs are produced by default, e.g. via global interceptors? How can I configure them? How to disable them?

Reply:
README updated.

Reply:
I have troubles finding answers to the 3 questions about the default behavior. Can you please point me to relevant place in README?

> - What formatters are registered by default? How can I configure them? How to disable them?
> - What transports are registered by default? How can I configure them? How to disable them?
> - What logs are produced by default, e.g. via global interceptors? How can I configure them? How to disable them?

Reply:
See https://github.com/strongloop/loopback-next/blob/logging/extensions/logging/README.md#configure-the-logging-component